spec/src/main/asciidoc/messaging-spec.adoc

```diff
@@ -26,3 +26,7675 @@ include::license-efsl.adoc[]
+|===
+|Jakarta Messaging Object |Supports Concurrent Use
+|Destination |YES
+|QueueConnectionFactory |YES
+|QueueConnection |YES
+|QueueSession |NO
+|QueueSender |NO
+|QueueReceiver |NO
+|===
+
+Table 2‑5 Objects used in the domain-specific API for pub/sub messaging,
+showing which support concurrent use
+
+[cols=",",options="header",]
+|===
+|Jakarta Messaging Object |Supports Concurrent Use
+|Destination |YES
+|TopicConnectionFactory |YES
+|TopicConnection |YES
+|TopicSession |NO
+|TopicPublisher |NO
+|TopicSubscriber |NO
+|===
+
+Jakarta Messaging defines some specific rules that restrict the 
+concurrent use of sessions. These apply to the Session object in the 
+classic API and to the QueueSession and TopicSession objects in the 
+domain-specific APIs. They also apply to the JMSContext object in the 
+simplified API since it encompasses a session. Since these rules require
+ more knowledge of Jakarta Messagingspecifics than we have presented at 
+this point, they will be described later. Here we will describe the 
+rationale for imposing them. 
+There are two reasons for restricting concurrent access to sessions.
+First, sessions are the Jakarta Messaging entity that supports 
+transactions. It is very difficult to implement transactions that are 
+multi-threaded. Second, sessions support asynchronous message 
+consumption. It is important that Jakarta Messaging _not_ require that 
+client code used for asynchronous message consumption be capable of 
+handling multiple, concurrent messages. In addition, if a session has 
+been set up with multiple, asynchronous consumers, it is important that 
+the client is not forced to handle the case where these separate 
+consumers are concurrently executing. These restrictions make Jakarta 
+Messaging easier to use for typical clients. More sophisticated clients 
+can get the concurrency they desire by using multiple sessions. In the 
+classic API and the domain-specific APIs this means using multiple 
+session objects. In the simplified API this means using multiple 
+JMSContext objects. 
+=== Triggering clients
+
+Some clients are designed to periodically wake up and process messages
+waiting for them. A message-based application triggering mechanism is
+often used with this style of client. The trigger is typically a
+threshold of waiting messages, etc.
+
+Jakarta Messaging does not provide a mechanism for triggering the 
+execution of a client. Some providers may supply such a triggering 
+mechanism via their administrative facilities.
+
+=== Request/reply
+
+Jakarta Messaging provides the JMSReplyTo message header field for 
+specifying the Destination where a reply to a message should be sent. 
+The JMSCorrelationID header field of the reply can be used to reference 
+the original request. See Section “” for more information.
+
+In addition, Jakarta Messaging provides a facility for creating 
+temporary queues and topics that can be used as a unique destination for
+ replies.
```

Comment:
Another gratuitous space at the beginning.  Is this the result of the line length tool?

Reply:
> Another gratuitous space at the beginning. Is this the result of the line length tool?

Probably. Thanks a lot for reviewing it.